uiowa/uiowa
# [Alerts] uids_base alerts 4.x update

Resolves https://github.com/uiowa/uiowa/issues/6790. 

# How to test

- Code review 
- Code review `alert.html.twig` template
- Review `aria-label=` and `role=` implementation on each alert. 

```
ddev blt frontend
```

## Test inline WYSIWYG alerts and update hook
```
ddev blt ds --site=sandbox.uiowa.edu && ddev drush @sandbox.local uli /kitchen-sink/
```
- Update hook code review
- Test update hook updating WYSIWYG alert classes on https://sitenow.uiowa.ddev.site/kitchen-sink and https://sitenow.uiowa.ddev.site/alerts-inline
- Create a new page and create new alerts using the WYSIWYG and confirm that new alert classes are applied. 

## Test layout builder inline alerts
- Edit a page layout and move blocks around and confirm that alert messages appear at the top of the page styled with the new alert styles and close button.
- Test other visible Drupal status messages that you can think of.

## Test sitewide custom `uiowa_alert`
```
ddev blt ds --site=sitenow.uiowa.edu && ddev drush @sitenow.local uli /admin/config/sitenow/uiowa-alerts
```
- Add sitewide alert and test all alert levels.  Confirm that color and icon changes for each selected alert level. 
- Edit a page layout and confirm that the `uiowa_alert` is not displayed at the top of the layout.
- Go to https://sitenow.uiowa.ddev.site/request-site in anonymous window and confirm that login alert is styled.
## Test webform alert and error messages

```
ddev blt ds --site=sitenow.uiowa.edu && ddev drush @sitenow.local uli /contact-us
```

- Go to https://sitenow.uiowa.ddev.site/contact-us and confirm that the form error message is 
  - Displayed below the `uiowa_alert` message.
  - Styled using the new alert classes.

```
ddev blt ds --site=sandbox.uiowa.edu && ddev drush @sandbox.local uli /form/joe-test-webform
```

- Go to https://sandbox.uiowa.ddev.site/form/joe-test-webform and test out the alert at the top of the form. 


## Test hawk alert
```
ddev blt ds --site=default && ddev drush @default.local cset uiowa_alerts.settings hawk_alert.source 'https://emergency.stage.drupal.uiowa.edu/api/active.json' -y && ddev drush @default.local uli node/1
```
- Confirm that Hawk alert is rendered in danger style alert. 
- Test with uiowa_alert custom message `/admin/config/sitenow/uiowa-alerts`
- Code review to confirm that none of the aria work has been removed from the hawk alert. 

## Test facilities alerts
```
ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local uli /building/0374
```
- Confirm that alert appears styled with warning alert style at the bottom of the page.
